protocols/dcutr/src/behaviour.rs

```diff
@@ -104,9 +104,7 @@ impl Behaviour {
             ..
         }: DialFailure,
     ) {
-        let peer_id = if let Some(peer_id) = peer_id {
-            peer_id
-        } else {
+        let Some(peer_id) = peer_id else {
```

Comment:
I am surprised the lint doesn't fire for any of the below ones? It looks like we should be able to use `let else` for some more of the cases here.

Reply:
Yes, clippy doesn't catch this flavor of `if let else` yet. Theres an open issue regarding this specific lint here: https://github.com/rust-lang/rust-clippy/issues/9807

I'll try to do some manual searching to see if theres more opportunities to refactor. And when the lint does become available, it should definitely be added to rust-libp2p. If I have some spare time I might take a stab at implementing it myself

Reply:
At a quick glance, there seems to be a bunch of these in the codebase. I'm happy to go through and make the changes, but its potentially a thousand+ lines of code. Is that ok?

Reply:
> At a quick glance, there seems to be a bunch of these in the codebase. I'm happy to go through and make the changes, but its potentially a thousand+ lines of code. Is that ok?

Yes but let's land them in separate PRs perhaps? Then the risk of merge conflicts is lower :)

If it is not (yet) caught by the lint, there is not reason to force it into this PR because CI will be green either way :)